integration_test/validate_metadata_test.go

```diff
@@ -28,6 +31,27 @@ func TestValidateMetadataOfThirdPartyApps(t *testing.T) {
 	}
 }
 
+func TestMissingMetadataForThirdPartyApps(t *testing.T) {
+	parentDirectory := "third_party_apps_data/applications"
+	dirs, err := thirdPartyDataDir.ReadDir(parentDirectory)
+
+	if err != nil {
+		t.Error(err)
+	}
+
+	for _, dir := range dirs {
+		dirName := dir.Name()
+		dirPath := path.Join(parentDirectory, dirName)
+		if _, er := os.Stat(path.Join(dirPath, "metadata.yaml")); errors.Is(er, os.ErrNotExist) {
+			err = multierr.Append(err, er)
```

Comment:
The logic here is a bit tricky. `err` will be `nil` at the start of the loop, because non-`nil` errors are reported earlier, but it takes some reasoning to arrive at that conclusion. I would clear the error explicitly here.

Also, what about errors other than `ErrNotExist`? Are they just silently swallowed?

Finally, `er` looks like a typo at first glance. Is this really what the Go style encourages?

Reply:
We can probably just raise the error without checking the error type here. In other words, as long as the test could not successfully read the `metadata.yaml` file, error out.

Reply:
Changed in: [fa22d58](https://github.com/GoogleCloudPlatform/ops-agent/pull/793/commits/fa22d58043929240fda65f8359a75886a5638c0a)

Reply:
Thanks!